pkg/crc/segment/segment.go

```diff
@@ -73,9 +74,9 @@ func (c *Client) Upload(action string, duration time.Duration, err error) error
 
 	properties := analytics.NewProperties().
 		Set("version", version.GetCRCVersion()).
-		Set("success", err == nil).
+		Set("success", err == nil || err.Error() == crcErr.VMNotExist.Error()).
```

Comment:
Does this have to be a string comparison? Could it be `err == crcErr.VMNotExist`? Feels a bit odd to add an error type, and not really use it, the way it's used, it could be replaced by `const VMNotExistStr = "Machine crc does not exist. Use 'crc start' to create it"`.
And it might make sense to record this separately, so that we can have reports for "command succeeded", "command did not succeed because the VM did not exist", "command failed" ?

Reply:
> Does this have to be a string comparison? Could it be `err == crcErr.VMNotExist`?

@cfergeau @guillaumerose We can't use either `err == crcErr.VMNotExist` or even `error.As()` because what we do as part of render we don't propagate same error but only propagate the error string and then check again if string is non empty, create a new one which break the error chain and show that user :(

- https://github.com/code-ready/crc/blob/master/cmd/crc/cmd/delete.go#L61
- https://github.com/code-ready/crc/blob/master/cmd/crc/cmd/delete.go#L74

>  Feels a bit odd to add an error type, and not really use it, the way it's used, it could be replaced by `const VMNotExistStr = "Machine crc does not exist. Use 'crc start' to create it"`.

I know, I am thinking a way to persist the error chain so we can do something like `error.As()` . It does feel old right now but it should be useful but if you want in I can change it to the string constant for time being.

> And it might make sense to record this separately, so that we can have reports for "command succeeded", "command did not succeed because the VM did not exist", "command failed" ?

The way I think for a user this command is failed since the vm is not exist but we really don't want to capture that data (are we?) what it will tell us in long run?

Reply:
Yeah, ideally we would not lose the initial error so that we can use `errors.As` or such. But no idea how involved this would be :)